src/main/scala/org/constellation/snapshotstreaming/App.scala

```diff
@@ -1,328 +1,66 @@
-      Stream
-        .emit(height.toString)
-        .through(fs2.text.utf8Encode)
-        .through(fs2.io.file.writeAll(Paths.get(configuration.lastSentHeightPath), blocker))
-    }
-
-  private def getLastStreamedHeightFromFile[F[_]: Concurrent: ContextShift]: Stream[F, Long] =
-    Stream.resource(Blocker[F]).flatMap { blocker: Blocker =>
-      fs2.io.file
-        .readAll(Paths.get(configuration.lastSentHeightPath), blocker, 4096)
-        .through(text.utf8Decode)
-        .through(text.lines)
-        .map(_.toLong)
-        .evalTap(
-          height =>
-            LiftIO[F]
-              .liftIO(
-                logger
-                  .info(s"Found last streamed height in file: $height. Returning starting height: ${height + 2L}.")
-              )
-        )
-        .map(_ + 2L)
-        .take(1)
-    }
-
-  private def putToES[F[_]: ConcurrentEffect: Timer: RaiseThrowable: Parallel](
-    elasticSearchDAO: ElasticSearchDAO[F]
-  )(result: S3DeserializedResult): Stream[F, Unit] =
-    elasticSearchDAO
-      .mapAndSendToElasticSearch(result)
-      .flatMap(
-        _ =>
-          Stream.eval(
-            LiftIO[F]
-              .liftIO(logger.info(s"[-> ES] ${result.height} OK"))
-          )
-      )
-      .handleErrorWith(
-        e =>
-          Stream.eval[F, Unit](
-            LiftIO[F].liftIO(
-              logger
-                .error(e)(s"[-> ES] ${result.height} ERROR")
-            )
-          )
-      )
-
-  private def retryInfinitely[F[_]: Concurrent: Timer, A](
-    delay: FiniteDuration
-  )(stream: Stream[F, A]): Stream[F, A] =
-    stream.attempts {
-      Stream
-        .unfold(delay)(d => Some(d -> delay))
-        .covary[F]
-    }.takeThrough(_.fold(scala.util.control.NonFatal.apply, _ => false))
-      .last
-      .map(_.get)
-      .rethrow
-
-  private def getHeightsFromFile[F[_]: Concurrent: ContextShift](
-    path: String
-  ): Stream[F, Long] =
-    Stream.resource(Blocker[F]).flatMap { blocker: Blocker =>
-      fs2.io.file
-        .readAll(Paths.get(path), blocker, 4 * 4096)
-        .through(text.utf8Decode)
-        .through(text.lines)
-        .filter(_.nonEmpty)
-        .map(_.toLong)
-    }
-
-  private def getHeights[F[_]: Concurrent](
-    startingHeight: Long,
-    snapshotInterval: Long = 2L,
-    endingHeight: Option[Long] = None
-  ): Stream[F, Long] = {
-    val heightsIterator = Stream.iterate(startingHeight)(_ + snapshotInterval)
-    endingHeight.fold(heightsIterator)(
-      endingHeight => heightsIterator.takeWhile(_ <= endingHeight)
-    )
-  }
-}
+      nodeClient = NodeClient.make(client)
+      lastOrdinalService = LastOrdinalService.make(configuration)
+      snapshotService = SnapshotService.make(nodeClient, lastOrdinalService, configuration)
+
+      res <- snapshotService.getSnapshots(None).evalTap(s => Logger[F].info(s.toString))
+      _ <- saveLastOrdinalToFile(res.ordinal.value)
+    yield res
+    end for
```

Comment:
With scala3 and such for syntax, I'd suggest writing it like that, but it's up to you:
```scala
def mkClient =  Stream.resource(
         EmberClientBuilder
           .default[F]
           .withTimeout(configuration.httpClientTimeout)
           .withIdleTimeInPool(configuration.httpClientIdleTime)
           .build

val lastOrdinalService = LastOrdinalService.make(configuration)
          
mkClient.flatMap { client =>
  val nodeClient = NodeClient.make(client)
  val snapshotService = SnapshotService.make(nodeClient, lastOrdinalService, configuration)
  
  snapshotService.getSnapshots(None)
}
.flatTap { s => Logger[F].info(s.toString) }
.flatTap { saveLastOrdinalToFile(_.ordinal.value) )
```